experimental/Schemes/BlowupMorphism.jl

```diff
@@ -96,108 +97,187 @@ function strict_transform(p::BlowupMorphism, inc::CoveredClosedEmbedding)
+  phi = (CY === default_covering(Y) ? Y[CY_simp,CY] : identity_map(CY_simp))
+  p_cov = compose(phi,p_cov_temp)    # blow up using simplified covering
 
-  # compose the covering morphisms
-  p_cov_simp = compose(phi, p_cov)
-  CX = codomain(p_cov)
-  E = exceptional_divisor(p)
+  ## do the transform on the charts
+  b = -2                               # safe initialization of multiplicity return value
+  bmin = -2                            # safe initialization of minimal multiplicity for weak transform in different charts
   for U in patches(CY_simp)
-    p_res = p_cov_simp[U]
-    V = codomain(p_res)
-    J = I(V)
-    #g = maps_with_given_codomain(inc, V)
-    pbJ = ideal(OO(U), pullback(p_res).(gens(J)))
-    pbJ_sat = saturated_ideal(pbJ)
-    pbJ_sat = saturation(pbJ_sat, ideal(base_ring(pbJ_sat), lifted_numerator.(E(U))))
-    pbJ = ideal(OO(U), [g for g in OO(U).(gens(pbJ_sat)) if !iszero(g)])
-    if length(gens(pbJ))==0 
-        error("output of saturation invalid; faulty exceptional divisor?")
+    V = codomain(p_cov[U])             # affine patch on X
+    Iorig_chart = I(V)                 # I on this patch
+    Itotal_chart = ideal(OO(U), pullback(p_cov[U]).(gens(Iorig_chart)))
+                                       # total transform on Chart
+
+    ## don't try saturating with respect to an empty set
+    ## not expensive for Cartier divisor, GB is cached (?) after first computation
+    if is_one(IE(U))
+      ID[U] = Itotal_chart
+      continue
     end
-    ID[U] = pbJ
+
+    IE_chart = IE(U)
+
+    ## do different methods according to integer argument method
+    if method == -1
+      Itrans_chart,btemp = saturation_with_index(Itotal_chart, IE_chart)                      # strict
+      b = max(b,btemp)
+   elseif method == 0
+      Itrans_chart,btemp = Oscar.iterated_quotients(Itotal_chart,IE_chart, method)             # weak
+      if b == -2
+         b = btemp
+      end
+      bmin = min(b,btemp)
+    else
+      Itrans_chart,b = Oscar.iterated_quotients(Itotal_chart,IE_chart, method)                 # controlled
+    end
+    ID[U] = Itrans_chart
   end
 
-  I_trans = IdealSheaf(Y, ID, check=false) 
-  return I_trans
+  bmin == -2 || bmin==b || error("different multiplicities in different charts, use controlled transform with control ",bmin)
+  b > -2 || error("no patches in CY_simp!!!")
+  I_trans = IdealSheaf(Y,ID,check=false)
+  return I_trans,b
 end
 
+##########################################################################################################
+## Handle Cartier divisors separately, as ideal quotients are quotients of ring elements in this case
+##########################################################################################################
+@doc """
+  strict_transform(p::BlowupMorphism, C::EffectiveCartierDivisor)
+
+For a `BlowupMorphism`  ``p : Y → X`` and an `EffectiveCartierDivisor` ``C`` on ``X`` return the
+strict transform of ``C`` on ``Y``.
+"""
```

Comment:
```suggestion
@doc """
  strict_transform(p::BlowupMorphism, C::EffectiveCartierDivisor)

For a blow up morphism  ``p : Y → X`` and an effective Cartier divisor  ``C`` on ``X`` return the
strict transform of ``C`` on ``Y``.
"""
```
Since the types are visible in the docstring, we can write it as we would in a paper.

Reply:
Do you insist on me changing it everywhere?

Reply:
It is just a suggestion.